pkg/tsdb/grafana-pyroscope-datasource/query.go

```diff
@@ -97,41 +97,59 @@ func (d *PyroscopeDatasource) query(ctx context.Context, pCtx backend.PluginCont
 	}
 
 	if query.QueryType == queryTypeProfile || query.QueryType == queryTypeBoth {
-		g.Go(func() error {
-			logger.Debug("Calling GetProfile", "queryModel", qm)
-			prof, err := d.client.GetProfile(gCtx, qm.ProfileTypeId, qm.LabelSelector, query.TimeRange.From.UnixMilli(), query.TimeRange.To.UnixMilli(), qm.MaxNodes)
-			if err != nil {
-				span.RecordError(err)
-				span.SetStatus(codes.Error, err.Error())
-				logger.Error("Error GetProfile()", "err", err)
-				return err
-			}
+		if len(qm.SpanSelector) > 0 {
+			g.Go(func() error {
```

Comment:
Again seems like this whole part is just duplicating the same code, seems like we need to just choose a proper function call and everything else is the same.